tests/fixtures/optimizer/canonicalize.sql

```diff
@@ -82,6 +82,9 @@ DATE_ADD(CAST("x" AS DATE), 1, 'YEAR');
 DATE_ADD('2023-01-01', 1, 'YEAR');
 DATE_ADD(CAST('2023-01-01' AS DATE), 1, 'YEAR');
 
+DATE_ADD('2023-01-01', 1.0, 'YEAR');
```

Comment:
can the unit be like hour where floats might make sense?

Reply:
Might just do this for presto, which only allows bigint